plugin/resurrect/workspace_state.lua

```diff
@@ -5,7 +5,8 @@ local pub = {}
 ---restore workspace state
 ---@param workspace_state workspace_state
 ---@param opts? restore_opts
-function pub.restore_workspace(workspace_state, opts)
+---@param window? MuxWindow
+function pub.restore_workspace(workspace_state, opts, window)
```

Comment:
I have tried to think of ways the window parameter could be omitted an found that on gnome i am able to post toast notifications with the following:
```lua
wezterm.gui.gui_windows()[1]:toast_notification('wezterm', 'test', nil, 3000)
```

If this also works on other desktop envoirements, then I would prefer this instead of adding an extra parameter.
If however it does not work, then I think the window parameter should be part of the `opts` table, i.e. `opts.window`

Reply:
I dont think you need to emit the window at all, as you can use that snippit for the toasts

Reply:
Nice find!
This works on MacOS too. No need to pass in a `window` object.